### ISSUE TITLE
Possible bug in 'directional coloured fog'

### ISSUE THREAD

**Issue by paramat** [minetest/minetest#5697]
```
	// Horizon coloring based on sun and moon direction during sunset and sunrise
	video::SColor pointcolor = video::SColor(255, 255, 255, m_bgcolor.getAlpha());
```
But SColor is ARGB.
SColorf is RGBA.
@MirceaKitsune 
Unfortunately cloud colours at dawn/dusk are currently broken due to a separate issue so keep this in mind. The effect of this error may be subtle, there is no obvious visual problem.

**Comment by kilbith** [comment]
Indeed, that code is wrong and needs a fix. `SColor` has [ARGB format](http://irrlicht.sourceforge.net/docu/classirr_1_1video_1_1_s_color.html#ab001c0c2515ff5f8a0b324d1345485f4).

**Comment by paramat** [comment]
Possibly the effect is not too large because the alpha of bgcolor is probably near or equal to 255.

**Comment by paramat** [comment]
Fixed in 2nd commit of #5693

**Comment by paramat** [comment]
#5693 merged.
